### DISCUSSION TITLE
Drop a melee weapon while it is hitting will not unregister its collision detection

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#14372] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When you pick up the melee weapon, it becomes active and handles all collisions for objects that collided with it.

https://github.com/user-attachments/assets/c33277e3-cecc-4257-b9c1-32513ba20cbe

### Reproduction steps

1. Swing a boarding axe
2. Drop the axe before it finishes its swing
3. Control a character and collide with the axe
4. Pick up the axe

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
The issue has now been fixed in our private repo, and the fix will be included in the next unstable build and the next update. Sorry for the inconvenience!
